src/instance.rs

```diff
@@ -97,6 +97,220 @@ where
     }
 }
 
+impl<'py, T> Bound<'py, T> {
+    /// Cast this to a concrete Python type or pyclass.
```

Comment:
I wonder if it's worth calling out here that unlike Python's `typing.cast`, this does have a runtime cost?

Reply:
Fair enough, even if we don't make the direct comparison it might be worth documenting that this performs a runtime type check (even if the astute reader might notice that from the `PyTypeCheck` trait bound).

Reply:
Sure thing, I've added a sentence mentioning that.